src/Symfony/Component/HttpFoundation/Tests/Session/Storage/Handler/Fixtures/storage.expected

```diff
@@ -11,10 +11,11 @@ $_SESSION is not empty
 write
 destroy
 close
-$_SESSION is not empty
+$_SESSION is empty
```

Comment:
So basically that means we "commit" the empty state.
We could, in theory, commit that when calling "get" on the flashbag, but this might be hard with the current design, and undesired as it could lead to another session being opened if the session object is used just after, right?

Reply:
I don't quite understand whyt you mean by "commit" but I guess you mean "storing the session".
So yes, that's what I find weird about what we do now.
So here's the two cases:

1. Either you have a session value - say `$_SESSION['foobar']` (so on top of the SF bags):

In that case, we remove the bags and we call the handler with only one value in the session. After that, we restore `$_SESSION` from `$session` which means we again have the `foobar` key plus all our bag keys. So in case anybody would access `$_SESSION` after calling `$session->save()` (which is nonsense anyway) it would represent the correct values.

2. Or you have no additional session values on top of the SF bags - aka the session is empty

In that case we, again, remove our bags, call the handler which then destroys the session because the session is empty and then we still restore `$_SESSION` which means even though we saved (or committed) an empty session, accessing `$_SESSION` after `$session->save()` will still have the bag keys. That's nonsense imho.